### PR TITLE
chore(deps): update boringssl version

### PR DESCRIPTION
**Pull request by coryan**
Part of the work for #10254

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10721)

### CODE REVIEW

File: bazel/google_cloud_cpp_deps.bzl [hunk 1]
```diff
@@ -112,17 +112,18 @@ def google_cloud_cpp_deps():
             sha256 = "22fdaf641b31655d4b2297f9981fa5203b2866f8332d3c6333f6b0107bb320de",
         )
 
-    # Load BoringSSL, this is used by gRPC, but as I write this (2021-06-03, circa gRPC-1.37.1), the version used by
-    # gRPC does not compile with GCC-11
+    # Load BoringSSL, this can be automatically loaded by gRPC. As-of 2023-02-01
```
Comment (devbww) [inline bazel/google_cloud_cpp_deps.bzl:115]: s/, this/. This/

File: bazel/google_cloud_cpp_deps.bzl [hunk 2]
```diff
@@ -112,17 +112,18 @@ def google_cloud_cpp_deps():
...
+    # the version loaded by gRPC-1.51 does not compile with Clang-15.
     if "boringssl" not in native.existing_rules():
         http_archive(
             name = "boringssl",
             # Use github mirror instead of https://boringssl.googlesource.com/boringssl
-            # to obtain a boringssl archive with consistent sha256
-            sha256 = "f0f433106f98f6f50ed6bbc2169f7c42dd73d13d0f09d431082519b5649903a6",
-            strip_prefix = "boringssl-56eb68f64215738552be452e311da12047934ab4",
+            # to obtain a boringssl archive with (more) consistent sha256
```
Comment (devbww) [inline bazel/google_cloud_cpp_deps.bzl:121]: s/with/with a/
s/$/./

**Review by devbww — APPROVED** [review]

File: bazel/google_cloud_cpp_deps.bzl (hunk 1 above)
Comment (coryan) [reply bazel/google_cloud_cpp_deps.bzl]: Done, with tweaks.

File: bazel/google_cloud_cpp_deps.bzl (hunk 2 above)
Comment (coryan) [reply bazel/google_cloud_cpp_deps.bzl]: Fixed.

**Pull request merged**